tests/test_cylindrical_geometry.py

```diff
@@ -0,0 +1,28 @@
+from pytest import approx, raises
+from PySide2.QtGui import QVector3D
+from .helpers import create_nexus_wrapper, add_component_to_file
+
```

Comment:
I think in general there should be more tests in here, for example:
* Changing properties of the cylinder
* Loading a cylinder from a file where it has multiple cylinders or cylinders that are not `[0,1,2]`
* That the OFF geometry  of the cylinder is correctly computed
* That the underlying nexus file changes correctly
* Probably many others, worth having a think

Reply:
The setters were not being used, so I've removed them. When we can edit components a cylinder may need to change, but they are as cheap to remove and recreate as to change, so I think keeping them immutable is more simple.

I've added various tests, including for out-of-order vertices and to document what happens if there are multiple cylinders defined in the group in file.

I've not tested the OFF geometry output for cylinders as this is not a regression for these changes and that functionality will soon be replaced: #397 (I don't think it is worth my time.)